openml/extensions/sklearn/extension.py

```diff
@@ -1512,10 +1515,11 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
             if not isinstance(classes, list):
                 raise ValueError('please convert model classes to list prior to '
                                  'calling this fn')
-            result = np.zeros((len(y), len(classes)), dtype=np.float32)
-            for obs, prediction_idx in enumerate(y):
-                result[obs][prediction_idx] = 1.0
-            return result
+            # DataFrame allows more accurate mapping of classes as column names
```

Comment:
I don't understand this change. What means more accurate here?

Reply:
Some target columns were being loaded as the class labels themselves. With a dataframe, having the class labels as column names allows an easier access using `loc`. With _accurate_ I meant the more foolproof access that is possible with dataframes.

Reply:
Sorry, I don't understand the original problem. Could you please give an example?